src/control/events/ras.go

```diff
@@ -127,6 +129,20 @@ type RASEvent struct {
 	ObjID        string          `json:"obj_id"`
 	CtlOp        string          `json:"ctl_op"`
 	ExtendedInfo RASExtendedInfo `json:"extended_info"`
+
+	forwarded bool
+}
+
+// IsForwarded returns true if event has been forwarded between hosts.
+func (evt *RASEvent) IsForwarded() bool {
+	return evt.forwarded
+}
+
+// WithIsForwarded sets the forwarded state of this event.
+func (evt *RASEvent) WithIsForwarded(isForwarded bool) *RASEvent {
+	evt.forwarded = isForwarded
+
+	return evt
```

Comment:
As long as there is no read of evt.forwarded before the write (and then of course there are no subsequent writes), I think that this should be safe to use concurrently with no synchronization. I would have to take a closer look to see if that's the case, though.

Reply:
addressed by changing forwarded to an atomic bool, see https://github.com/daos-stack/daos/pull/4435